src/bulk/common.ts

```diff
@@ -679,11 +679,22 @@ function handleMongoWriteConcernError(
  */
 export class MongoBulkWriteError extends MongoServerError {
   result: BulkWriteResult;
+  writeErrors: OneOrMore<WriteError> = [];
+  err?: WriteConcernError;
 
   /** Creates a new MongoBulkWriteError */
-  constructor(error: AnyError, result: BulkWriteResult) {
-    super(error as Error);
-    Object.assign(this, error);
+  constructor(
+    x: { message: string; code: number; writeErrors?: WriteError[] } | WriteConcernError | AnyError,
+    result: BulkWriteResult
+  ) {
+    super(x);
+    if (x instanceof Error) Object.assign(this, x);
```

Comment:
I believe we put the assign behind the gate of `instanceof Error` since that was filtering for MongoServerErrors while testing. 
We should move the assign to be the last call of the constructor ungated. (Or remove it all together and see if tests fail, but I defer to the team on seeing if that's acceptable).